internal/pkg/table/policy.go

```diff
@@ -3586,16 +3593,19 @@ func (r *RoutingPolicy) DeleteStatement(st *Statement, all bool) (err error) {
 }
 
 func (r *RoutingPolicy) GetPolicy(name string) []*config.PolicyDefinition {
-	r.mu.RLock()
+	ps := func() Policies {
+		r.mu.RLock()
+		defer r.mu.RUnlock()
```

Comment:
This one is not fixing an existing dead lock. However, I like using the pattern of immediately deferring the unlock after locking. That way, if a future code changes introduces a return statement or something, there is no chance that we can forget to unlock the lock. To maintain the scope of the lock the same as the original, I added this anonymous function.